File: 30.2/ubuntu/24.04/x86_64/systems-ide/flight-tests/rust/src/counter.rs
/// A simple counter, exercising hover docs, go-to-definition, and rename
/// across files (main.rs -> counter.rs).
#[derive(Debug)]
pub struct Counter {
    pub n: i32,
    pub name: String,
}

impl Counter {
    pub fn inc(&mut self) {
        self.n += 1;
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn increments_by_one() {
        let mut c = Counter {
            n: 0,
            name: "test".to_string(),
        };
        c.inc();
        assert_eq!(c.n, 1);
    }
}

File: 30.2/ubuntu/24.04/x86_64/systems-ide/flight-tests/zig/src/counter.zig
const std = @import("std");

/// A simple counter, exercising hover docs, go-to-definition, and rename
/// across files (main.zig -> counter.zig).
pub const Counter = struct {
    n: i32,
    name: []const u8,

    pub fn inc(self: *Counter) void {
        self.n += 1;
    }
};

test "increments by one" {
    var c = Counter{ .n = 0, .name = "test" };
    c.inc();
    try std.testing.expectEqual(@as(i32, 1), c.n);
}
